test/cpp/test_ir.cpp

```diff
@@ -94,5 +95,91 @@ TEST(IrTest, TestScopePusherWithDebugging) {
   FLAGS_torch_lazy_ir_debug = restore_FLAGS_torch_lazy_ir_debug;
 }
 
+TEST(IrTest, TestSizeNode) {
+  torch::lazy::NodePtr scalar_node =
+      ScalarOp(1.0, xla::ShapeUtil::MakeShape(xla::F32, {3, 4}));
+  torch::lazy::NodePtr size_node_0 =
+      torch::lazy::MakeNode<SizeNode>(scalar_node, 0);
+  torch::lazy::NodePtr size_node_1 =
+      torch::lazy::MakeNode<SizeNode>(scalar_node, 1);
+  std::shared_ptr<torch::lazy::DimensionNode> dim_node_0 =
+      std::dynamic_pointer_cast<torch::lazy::DimensionNode>(size_node_0);
+  std::shared_ptr<torch::lazy::DimensionNode> dim_node_1 =
+      std::dynamic_pointer_cast<torch::lazy::DimensionNode>(size_node_1);
+
+  EXPECT_EQ(dim_node_0->getStaticValue(), 3);
+  EXPECT_EQ(dim_node_1->getStaticValue(), 4);
+
+  ForEachDevice([&](const torch::lazy::BackendDevice& device) {
+    // Lower the SizeNode and execute the GetDimensionSize.
+    auto results = ExecuteAndFetch({size_node_0, size_node_1}, device);
+    EXPECT_EQ(results[0].sum().item().toInt(), 3);
+    EXPECT_EQ(results[1].sum().item().toInt(), 4);
+  });
+}
+
+TEST(IrTest, TestSizeAddNode) {
+  torch::lazy::NodePtr scalar_node =
+      ScalarOp(1.0, xla::ShapeUtil::MakeShape(xla::F32, {3, 4}));
+  torch::lazy::NodePtr size_node_0 =
+      torch::lazy::MakeNode<SizeNode>(scalar_node, 0);
+  torch::lazy::NodePtr size_node_1 =
+      torch::lazy::MakeNode<SizeNode>(scalar_node, 1);
+  torch::lazy::NodePtr size_node_add =
+      torch::lazy::MakeNode<SizeAdd>(size_node_0, size_node_1);
+  std::shared_ptr<torch::lazy::DimensionNode> dim_node_add =
+      std::dynamic_pointer_cast<torch::lazy::DimensionNode>(size_node_add);
+
+  EXPECT_EQ(dim_node_add->getStaticValue(), 7);
+
+  ForEachDevice([&](const torch::lazy::BackendDevice& device) {
+    // Lower the SizeAddNode and execute the GetDimensionSize.
+    auto results = ExecuteAndFetch({size_node_add}, device);
```

Comment:
do you think `ExecuteAndFetch` will be used to implement for `getDynamicValue()` or is this for testing?

Reply:
`ExecuteAndFetch` calls `GetTensors` which is what I used to implement `getDynamicValue`